File: Hard/median_of_two_sorted_arrays.php
<?php

class Solution {

    /**
     * @param Integer[] $nums1
     * @param Integer[] $nums2
     * @return Float
     */
    function findMedianSortedArrays($nums1, $nums2) {
        if (count($nums1) > 1000 || count($nums2) > 1000) {
		    return 0;
	    }

        $joinArr = array_merge($nums1, $nums2);
        sort($joinArr);
        $lenJoinArr = count($joinArr);

        if ($lenJoinArr == 0 || $lenJoinArr > 2000) {
            return 0;
        }

        $mid = $lenJoinArr / 2;
        if ($lenJoinArr % 2 == 1) {
            return floatval($joinArr[$mid]);
        }

        return floatval(($joinArr[$mid-1] + $joinArr[$mid]) / 2.0);
    }
}

$solution = new Solution;
// print_r($solution->findMedianSortedArrays([1,3], [2]));
// print_r($solution->findMedianSortedArrays([1,2], [3,4]));
print_r($solution->findMedianSortedArrays([1,2,3,4,5], [6,7,8,9,10,11,12,13,14,15,16,17]));

File: Hard/median_of_two_sorted_arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        if len(nums1) > 1000 or len(nums2) > 1000:
            return None
        
        joinArr = sorted(nums1 + nums2)
        lenJoinArr = len(joinArr)
        
        if lenJoinArr == 0 or lenJoinArr > 2000:
            return None
        
        mid = lenJoinArr // 2
        if lenJoinArr % 2 == 1:
            return float(joinArr[mid])
        else:
            return (joinArr[mid - 1] + joinArr[mid]) / 2.0
        
        
        
        
print(Solution().findMedianSortedArrays([1,3], [2]))
print(Solution().findMedianSortedArrays([1,2], [3,4]))